src/rgw/rgw_user.cc

```diff
@@ -2299,7 +2299,7 @@ int RGWUserAdminOp_User::info(RGWRados *store, RGWUserAdminOpState& op_state,
     return ret;
 
   if (!op_state.has_existing_user())
-    return -ENOENT;
+    return -ERR_NO_SUCH_USER;
```

Comment:
What about ` if (!op_state.has_existing_subuser()) ` ? I think similar improvement can be implemented for this too.